content/en/docs/tutorials/configuration/updating-configuration-via-a-configmap.md

```diff
@@ -0,0 +1,381 @@
+{{% code_sample file="deployments/deployment-with-configmap-two-containers.yaml" %}}
+
+Create the deployment.
+```shell
+kubectl apply -f https://k8s.io/examples/deployments/deployment-with-configmap-two-containers.yaml
+```
+
+Check the deployment.
+```shell
+kubectl get deployment configmap-two-containers
+```
+
+You should see an output similar to:
+```shell
+NAME                       READY   UP-TO-DATE   AVAILABLE   AGE
+configmap-two-containers   3/3     3            3           12s
+
+```
+
+Check the pods.
+```shell
+kubectl get pods --selector=app=configmap-two-containers
+```
+You should see an output similar to:
+```shell
+NAME                                        READY   STATUS    RESTARTS   AGE
+configmap-two-containers-565fb6d4f4-2xhxf   2/2     Running   0          20s
+configmap-two-containers-565fb6d4f4-g5v4j   2/2     Running   0          20s
+configmap-two-containers-565fb6d4f4-mzsmf   2/2     Running   0          20s
+````
+
+Expose the deployment via a Service.
+```shell
+kubectl expose deployment configmap-two-containers --name=configmap-service --port=8080 --target-port=80
+```
+
+Use ```kubectl``` to forward the port:
+```shell
+kubectl port-forward service/configmap-service 8080:8080 &
+```
+
+Access the service.
+```shell
+curl localhost:8080
+```
+
+You should see an output similar to:
+```shell
+Fri Jan  5 08:08:22 UTC 2024 My preferred colour is red
+```
+
+Edit the ConfigMap.
+```shell
+kubectl edit configmap colour
+```
+
+In the editor that appears, change the value of key `colour` from `red` to `blue`. Save your changes.
+The kubectl tool updates the ConfigMap accordingly (if you see an error, try again).
+
+Here's an example of how that manifest could look after you edit it:
+
+```yaml
+apiVersion: v1
+data:
+  colour: blue
+kind: ConfigMap
+# You can leave the existing metadata as they are.
+# The values you'll see won't exactly match these.
+metadata:
+  creationTimestamp: "2024-01-05T08:12:05Z"
+  name: colour
+  namespace: configmap
+  resourceVersion: "1801272"
+  uid: 80d33e4a-cbb4-4bc9-ba8c-544c68e425d6
+```
+
+Loop over the service URL for few seconds. 
+```shell
+while true; do curl localhost:8080; sleep 10; done
+```
+
+You should see the output change as follows:
+```shell
+Fri Jan  5 08:14:00 UTC 2024 My preferred colour is red
+Fri Jan  5 08:14:02 UTC 2024 My preferred colour is red
+Fri Jan  5 08:14:20 UTC 2024 My preferred colour is red
+Fri Jan  5 08:14:22 UTC 2024 My preferred colour is red
+Fri Jan  5 08:14:32 UTC 2024 My preferred colour is blue
+Fri Jan  5 08:14:43 UTC 2024 My preferred colour is blue
+Fri Jan  5 08:15:00 UTC 2024 My preferred colour is blue
+```
+
```

Comment:
I will work on adding this, however it may need some time. Once its ready, we can add it to the current PR if the PR is still open. Else, I can create a new PR for it later.

I believe this should not be a blocker to merge the current PR.

Reply:
Added a 4th scenario with the Sidecar Container.